vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go

```diff
@@ -203,10 +206,15 @@ func (f *podsEvictionRestrictionFactoryImpl) NewPodsEvictionRestriction(pods []*
 	creatorToSingleGroupStatsMap := make(map[podReplicaCreator]singleGroupStats)
 
 	for creator, replicas := range livePods {
+		// Use per-VPA minReplicas if present, fall back to the global setting.
+		required := f.minReplicas
+		if vpa.Spec.UpdatePolicy != nil && vpa.Spec.UpdatePolicy.MinReplicas != nil {
+			required = int(*vpa.Spec.UpdatePolicy.MinReplicas)
+		}
 		actual := len(replicas)
-		if actual < f.minReplicas {
-			klog.V(2).Infof("too few replicas for %v %v/%v. Found %v live pods",
-				creator.Kind, creator.Namespace, creator.Name, actual)
+		if actual < required {
+			klog.V(2).Infof("too few replicas for %v %v/%v. Found %v live pods, needs %v",
```

Comment:
It might be useful to log global vs override too

Reply:
Added info here & separate log when choosing the value.